week-3/Homework/mandatory/7-password-validator.js

```diff
@@ -21,10 +21,54 @@ Expected Result:
 PasswordValidationResult=  [false, false, false, false, true]
 
 */
+// const previousPassword = [
+//   "Se%5", 
+//   "TktE.TJTU", 
+//   "384#HsHF", 
+//   "dvyyeyy!5", 
+//   "tryT3729",
+//   "StUFf27%", 
+//   "Pl3nty!", 
+//   "Jai33", 
+//   "shajsaUA**&&", 
+//   "Pl3nty!"
+// ]
+// function validatePasswords(passwords) {
+//   return passwords.map(function(password) {
+//     console.log(/^[a-z][A-Z][0-9][\!\#\$%\.]/.test("y!e7#TD345"));
+//   });
+// }
+// validatePasswords(previousPassword);
 
-function validatePasswords(passwords) {
-
-}
+function testPassword(testPassword, index, allPasswords ) {
+  // testPassword - element being checked; index - index of element; allPassword - array of strings that has passwords
+  //we know the current password, were going through checking the existing password
+  //current if it exists and if it is the same it shoud, refutn false, if we get to the end of the loop and there are no dupes then we continue 
+    
+    for (let i = 0; i < index; ++i) {
+      if(testPassword === allPasswords[i]){
+        return false
+      }
+    }
+    let checkLowercase= /[a-z]/
+    let checkUppercase= /[A-Z]/
+    let checkNumber= /[0-9]/
+    let checkSymbol= /[!#$%.]/
+  if(checkLowercase.test(testPassword) &&
+      checkUppercase.test(testPassword) &&
+      checkNumber.test(testPassword) &&
+      checkSymbol.test(testPassword) &&
+      testPassword.length >= 5
+      ){
+          return true
+        }  else {
+          return false
+        }
+  } 
+  function validatePasswords(arr){
+    let newArr = arr.map(testPassword)
```

Comment:
Like with the previous exercise, this works, but could just be `return arr.map(testPassword);` :)